packages/amplify-category-auth/src/provider-utils/awscloudformation/utils/generate-auth-trigger-template.ts

```diff
@@ -152,7 +153,7 @@ const createCustomResource = (
 ): void => {
   const triggerCode = fs.readFileSync(authTriggerAssetFilePath, 'utf-8');
   const authTriggerFn = new lambda.Function(stack, 'authTriggerFn', {
-    runtime: lambda.Runtime.NODEJS_18_X,
+    runtime: new lambda.Runtime('nodejs22.x', RuntimeFamily.NODEJS, { supportsInlineCode: true }),
```

Comment:
Node 22 is not available as enum in currently used version of CDK.

This is a workaround recommended here https://github.com/aws/aws-cdk/blob/8b472fc5a68c2c83349dbfc11fa502f08d0bd5c8/packages/aws-cdk-lib/aws-lambda/lib/runtime.ts#L50-L51 .

And matches https://github.com/aws/aws-cdk/blob/8b472fc5a68c2c83349dbfc11fa502f08d0bd5c8/packages/aws-cdk-lib/aws-lambda/lib/runtime.ts#L124